### PR TITLE
feat: use query_count as a FastAPI dependency

### PR DESCRIPTION
**Pull request by hadrien**
## Description

* Custom pagination relying on a callable rather than a FastAPI dependency is sub-optimal;
* This PR introduces a breaking change to change that callable to be a dependency;

### CODE REVIEW

File: tests/test_pagination.py [hunk 1]
```diff
@@ -217,15 +216,18 @@ def sqla_14_all_users_with_notes_count(paginate: Paginate = Depends()):
         )
         return paginate(query, scalars=False)
 
-    def query_count(session: Session, query: Select) -> int:
-        return session.execute(select(func.count()).select_from(user_cls)).scalar()
+    def count_user_notes(user_id: int, session: Session = Depends()) -> int:
+        return session.execute(
+            select(func.count(note_cls.id)).where(note_cls.user_id == user_id)
+        ).scalar()
```
Comment (hadrien) [inline tests/test_pagination.py:222]: @vicrep I strongly believe the query counter function should be a FastAPI dependency to support this ^ 

What do you think?

Comment (vicrep) [reply tests/test_pagination.py:222]: @hadrien I agree in the context of SQLA2 language. The nice thing though is that making it a dependency still allows us to do pure counter functions if DI is not needed (basically, if you don't need it to be a dependency, then pretend it isn't one :) )

File: tests/test_pagination.py [hunk 2]
```diff
@@ -217,15 +216,18 @@ def sqla_14_all_users_with_notes_count(paginate: Paginate = Depends()):
...
 
-    CustomPaginate: PaginateSignature = Pagination(query_count=query_count)
+    CustomPaginate: PaginateSignature = Pagination(query_count=count_user_notes)
```
Comment (vicrep) [inline tests/test_pagination.py:224]: can we type the return type of `Pagination` so that `PaginateSignature` is inferred?

**Review by vicrep — APPROVED** [review]
Me gusta

**Pull request merged**